notebook/static/notebook/js/codemirror-ipython.js

```diff
@@ -5,34 +5,35 @@
 
 (function(mod) {
   if (typeof exports == "object" && typeof module == "object"){ // CommonJS
-    mod(require("codemirror/lib/codemirror"),
-        require("codemirror/mode/python/python")
+    mod(require("codemirror/lib/codemirror")
         );
   } else if (typeof define == "function" && define.amd){ // AMD
-    define(["codemirror/lib/codemirror",
-            "codemirror/mode/python/python"], mod);
+    define(["codemirror/lib/codemirror"], mod);
   } else {// Plain browser env
     mod(CodeMirror);
   }
 })(function(CodeMirror) {
     "use strict";
 
-    CodeMirror.defineMode("ipython", function(conf, parserConf) {
-        var pythonConf = {};
-        for (var prop in parserConf) {
-            if (parserConf.hasOwnProperty(prop)) {
-                pythonConf[prop] = parserConf[prop];
+    requirejs(['codemirror/mode/python/python'], function () {
```

Comment:
One thing I'm worried about is that we **can** do that for the IPython mode, but some codemirror modes do have this pattern:

```
django/django.js:        require("../../addon/mode/overlay"));
dockerfile/dockerfile.js:    mod(require("../../lib/codemirror"), require("../../addon/mode/simple"));
factor/factor.js:    mod(require("../../lib/codemirror"), require("../../addon/mode/simple"));
gfm/gfm.js:    mod(require("../../lib/codemirror"), require("../markdown/markdown"), require("../../addon/mode/overlay"));
handlebars/handlebars.js:    mod(require("../../lib/codemirror"), require("../../addon/mode/simple"), require("../../addon/mode/multiplex"));
htmlembedded/htmlembedded.js:        require("../../addon/mode/multiplex"));
modelica/modelica.js:    mod(require("../../lib/codemirror"));
nsis/nsis.js:    mod(require("../../lib/codemirror"), require("../../addon/mode/simple"));
rst/rst.js:    mod(require("../../lib/codemirror"), require("../python/python"), require("../stex/stex"), require("../../addon/mode/overlay"));
rust/rust.js:    mod(require("../../lib/codemirror"), require("../../addon/mode/simple"));
tornado/tornado.js:        require("../../addon/mode/overlay"));
twig/twig.js:    mod(require("../../lib/codemirror"),  require("../../addon/mode/multiplex"));
vue/vue.js:        require("../../addon/mode/overlay"),
```

So they will likely be broken...